### PR TITLE
docs: add README in vendor dir to illustrate the vendor policy

### PR DESCRIPTION
**Pull request by allencloud**
Signed-off-by: Allen Sun <allensun.shl@alibaba-inc.com>

### Ⅰ. Describe what this PR did

add README in vendor dir to illustrate the vendor policy

This is a very good guide for developers to manage vendoring. Most of the added file is from project PouchContainer.

### Ⅱ. Does this pull request fix one issue?

none

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
no need, just doc related.

### Ⅳ. Describe how to verify it
none

### Ⅴ. Special notes for reviews
cc @zhouhaibing089 @Starnop

### CODE REVIEW

File: vendor/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Vendor Mechanism
```
Comment (zhouhaibing089) [inline vendor/README.md]: Is is more appropriate to put this into `CONTRIBUTING.md`?

Comment (allencloud) [reply vendor/README.md]: It seems what you said is also reasonable. Let me try that as well with amending the commits.

**Comment by zhouhaibing089** [comment]
The doc looks good!

**Comment by allencloud** [comment]
I have update the vendor mechanism doc to the Readme. @zhouhaibing089 @Starnop PTAL

**Comment by Starnop** [comment]
LGTM.

**Pull request merged**